app/code/Magento/InventoryLowStockNotification/Block/Adminhtml/Product/Lowstock/Grid/Options/SourceCodeOptionHash.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryLowStockNotification\Block\Adminhtml\Product\Lowstock\Grid\Options;
+
+use Magento\Framework\Option\ArrayInterface;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+
+class SourceCodeOptionHash implements ArrayInterface
+{
+    /**
+     * @var SourceRepositoryInterface
+     */
+    protected $sourceRepository;
+
+    /**
+     * SourceCodeOptionHash constructor.
+     *
+     * @param SourceRepositoryInterface $sourceRepository
+     */
+    public function __construct(
+        SourceRepositoryInterface $sourceRepository
+    ) {
+        $this->sourceRepository = $sourceRepository;
+    }
+
+    /**
+     * Return array of available sources
+     *
+     * @return array
+     */
+    public function toOptionArray(): array
+    {
+        $optionArray = [];
```

Comment:
We don't need to show all of the existed sources
Need to show only possible to choosing

But it can be hard to implement in legacy grid, so it is optional remark